cli/src/main/java/com/box/l10n/mojito/cli/command/CommandHelper.java

```diff
@@ -360,6 +381,61 @@ ZonedDateTime getLastWeekDateIfTrue(boolean condition) {
     return dateTime;
   }
 
+  private static String getTimeframesRegex() {
+    final StringJoiner timeframeRegexJoiner = new StringJoiner("|");
+    Arrays.stream(TimeframeType.values())
+        .forEach(
+            timeframeType ->
+                timeframeRegexJoiner
+                    .add(String.valueOf(timeframeType.getAbbreviationInUpperCase()))
+                    .add(String.valueOf(timeframeType.getAbbreviationInLowerCase())));
+    return String.format(TIMEFRAMES_REGEX, timeframeRegexJoiner);
+  }
+
+  private static Integer getTimeframe(String input, TimeframeType timeframeType)
+      throws CommandException {
+    String regex =
+        String.format(
+            "\\d+(?=[%c|%c])",
+            timeframeType.getAbbreviationInUpperCase(), timeframeType.getAbbreviationInLowerCase());
+    Pattern pattern = Pattern.compile(regex);
+    Matcher matcher = pattern.matcher(input);
+    int i = 0;
+    int timeframe = 0;
+    while (matcher.find()) {
+      if (i > 0) {
+        throw new CommandException("Duplicated timeframes are not allowed");
+      }
+      timeframe = Integer.parseInt(matcher.group());
+      i++;
+    }
+    return timeframe;
+  }
+
+  static ZonedDateTime getCreatedBeforeDateTime(final String timeframe) {
+    if (timeframe == null) {
+      return null;
+    }
+    if (!timeframe.matches(getTimeframesRegex())) {
+      throw new CommandException(
+          String.format(
+              "Please enter a valid timeframe %s", BRANCH_CREATED_BEFORE_OPTIONS_AND_EXAMPLE));
+    }
+    final List<Period> periods = new ArrayList<>();
+    Arrays.stream(TimeframeType.values())
+        .forEach(
+            timeframeType -> {
+              if (TIMEFRAME_FUNCTIONS.containsKey(timeframeType)) {
+                periods.add(
+                    TIMEFRAME_FUNCTIONS
+                        .get(timeframeType)
+                        .apply(getTimeframe(timeframe, timeframeType)));
+              }
+            });
+    final Period totalPeriod = periods.stream().reduce(Period.ZERO, Period::plus);
```

Comment:
Do we ever expect multiple periods to be specified or supported in order to do a reduce/add on them?

I'd expect only one type of period to be specified at any point in time - which means the code above could be simplified and only do a check for a findFirst() for a match on the timeframeType and use that without the need for reduce().